### PR TITLE
Refactor UTXO scan

### PR DESCRIPTION
**Pull request by k9ert**
This PR refactors the UTXO-scan so that it creates a `command` which for now instantiated and executed from the wallet. As we already had a command (roughly based on this [pattern](https://en.wikipedia.org/wiki/Command_pattern)), i refactored both files in a newly created `commands` directory.

Also i've changed the timeout of all calls done for the utxo-scan to 5 mins which is acceptable as this is called in a background thread.

In order to test this, i've created a fixture which delivers a `specter_testnet_configured` (similiar to a `specter_regtest_configured`) which assumes a testnet3 node running on localhost. The corresponding test is `pytest.mark.skip()` but for manual testing, it's quite useful. 

As this is now the third file containing fixtures, i renamed all of the with a `fix_` prefix.

### CODE REVIEW

File: tests/test_commands_utxo_scanner.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        # With an explorer, it should work on a pruned_node:
+        mycmd = UtxoScanner(wallet, explorer="https://mempool.space/testnet/")
+        mycmd.execute(asyncc=False)
+        utxos = wallet.full_utxo
+        assert len(utxos) == 6
+        assert_exact_utxo_set(utxos)
+
+    else:
+        # On a pruned_node, you don't need any explorer
```
Comment (moneymanolis) [inline tests/test_commands_utxo_scanner.py:46]:
```suggestion
        # With a full node, you don't need any explorer
```

File: src/cryptoadvance/specter/commands/utxo_scanner.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    ):
+        self.wallet = wallet
+        self.explorer = explorer
+        if self.explorer:
+            self.explorer = self.explorer.rstrip("/")
+        self.requests_session = requests_session
+        self.only_tor = only_tor
+        self.error_msgs = []
+
+    def execute(self, asyncc=False):
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/commands/utxo_scanner.py:32]: asyncc with two cc?

Comment (k9ert) [reply src/cryptoadvance/specter/commands/utxo_scanner.py:32]: Yes, as async is a reserved word in python. I still don't have a consistent pattern how to deal with such cases. That's somehow i come up with `claz` or `clas`.

File: src/cryptoadvance/specter/commands/utxo_scanner.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        if asyncc:
+            t = threading.Thread(
+                target=self._execute,
+            )
+            t.start()
+        else:
+            self._execute()
+
+    def _execute(self):
+        self.prepare_scantxoutset_args()
```
Comment (k9ert) [inline src/cryptoadvance/specter/commands/utxo_scanner.py:42]: self.arg ...

File: src/cryptoadvance/specter/commands/utxo_scanner.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        self.wallet.fetch_transactions()
+        self.wallet.check_addresses()
+
+    def prepare_scantxoutset_args(self):
+        self.args = [
+            "start",
+            [
+                {
+                    "desc": self.wallet.recv_descriptor,
+                    "range": max(self.wallet.keypool, 1000),
```
Comment (k9ert) [inline src/cryptoadvance/specter/commands/utxo_scanner.py:113]: UI feedback for the user would be good. Create separate issue.

File: src/cryptoadvance/specter/commands/utxo_scanner.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+        updated = False
+        if max_recv >= self.wallet.address_index:
+            # skip to max_recv
+            self.wallet.address_index = max_recv
+            logger.info(f"Adjusted address_index of {self.wallet} to {max_change}")
+            # get next
+            self.wallet.getnewaddress(change=False, save=False)
+            updated = True
+        while max_change >= self.wallet.change_index:
```
Comment (k9ert) [inline src/cryptoadvance/specter/commands/utxo_scanner.py:148]:
```suggestion
        if max_change >= self.wallet.change_index:
```

File: src/cryptoadvance/specter/commands/utxo_scanner.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        #   'amount': 0.02119209, 'height': 1939628
+        # }]
+        self.unspents = self.wallet.rpc.scantxoutset(*self.args, timeout=self.timeout)[
+            "unspents"
+        ]
+        logger.info(f"Found {len(self.unspents)} utxo")
+        self.adjust_keypools()
+        # Endgoal: calling importprunedfunds with the raw-transaction and a proof that the tx was included in a block
+        self.add_blockhashes_to_txs_in_unspents()
+        self.add_proofs_to_txs_in_unspents()
```
Comment (k9ert) [inline src/cryptoadvance/specter/commands/utxo_scanner.py:55]: Could be implemented faster for fullnodes ?!

File: tests/fix_testnet.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+"""
+
+from cryptoadvance.specter.user import User, hash_password
+import pytest
+from cryptoadvance.specter.specter import Specter
+from cryptoadvance.specter.user import User
+
+
+@pytest.fixture
+def specter_testnet_configured(bitcoin_regtest, devices_filled_data_folder):
```
Comment (k9ert) [inline tests/fix_testnet.py:12]:
```suggestion
def specter_testnet_configured(devices_filled_data_folder):
```

File: tests/fix_testnet.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    """This assumes a bitcoin-testnet-node is running on loalhost"""
+
+    config = {
+        "rpc": {
+            "autodetect": False,
+            "datadir": "",
+            "user": "bitcoin",
+            "password": "secret",
```
Comment (k9ert) [inline tests/fix_testnet.py:20]:
```suggestion
            "user": "bitcoin", # change this to your credential in bitcoin.conf (for testnet)
            "password": "secret",
```

**Review by moneymanolis — APPROVED** [review]

File: src/cryptoadvance/specter/commands/utxo_scanner.py (hunk 4 above)
Comment (k9ert) [reply src/cryptoadvance/specter/commands/utxo_scanner.py:113]: https://github.com/cryptoadvance/specter-desktop/issues/1695

File: src/cryptoadvance/specter/commands/utxo_scanner.py (hunk 6 above)
Comment (k9ert) [reply src/cryptoadvance/specter/commands/utxo_scanner.py:55]: @moneymanolis please create an issue if necessary

**Pull request merged**